lib/install/ova/configure.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ova
+
+import (
+	"context"
+	"net"
+	"net/url"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/lib/config/dynamic/admiral"
+	"github.com/vmware/vic/pkg/errors"
+	"github.com/vmware/vic/pkg/vsphere/session"
+	"github.com/vmware/vic/pkg/vsphere/tasks"
+	"github.com/vmware/vic/pkg/vsphere/vm"
+)
+
+const (
+	// ManagedByKey defines the extension key to use in the ManagedByInfo of the OVA
+	ManagedByKey = "com.vmware.vic"
+	// ManagedByType defines the type to use in the ManagedByInfo of the OVA
+	ManagedByType = "VicApplianceVM"
+)
+
+func configureManagedByInfo(ctx context.Context, sess *session.Session, v *vm.VirtualMachine) error {
+	spec := types.VirtualMachineConfigSpec{
+		ManagedBy: &types.ManagedByInfo{
+			ExtensionKey: ManagedByKey,
+			Type:         ManagedByType,
+		},
+	}
+
+	info, err := v.WaitForResult(ctx, func(ctx context.Context) (tasks.Task, error) {
+		return v.Reconfigure(ctx, spec)
+	})
+
+	if err != nil {
+		log.Errorf("Error while setting ManagedByInfo: %s", err)
+		return err
+	}
+
+	if info.State != types.TaskInfoStateSuccess {
+		log.Errorf("Setting ManagedByInfo reported: %s", info.Error.LocalizedMessage)
+		return err
+	}
+
+	return nil
+}
+
+// ConfigureManagedByInfo takes sets the ManagedBy field for the VM specified by ovaURL
+func ConfigureManagedByInfo(ctx context.Context, sess *session.Session, ovaURL string) error {
+	v, err := getOvaVMByTag(ctx, sess, ovaURL)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Attempting to configure ManagedByInfo")
+	err = configureManagedByInfo(ctx, sess, v)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Successfully configured ManagedByInfo")
+	return nil
+}
+
+func getOvaVMByTag(ctx context.Context, sess *session.Session, u string) (*vm.VirtualMachine, error) {
```

Comment:
The entrypoint into this code is via the ui install script. The user supplies username/password, and a new session is created in the `vic-ui` binary. That session is passed along into this function, whose primary purpose is to verify that the host/IP of the VM we found via the tag is the same host/IP as specified in the OVA url that we passed in (the OVA url is auto-populated into the installer script on the backend when the "Getting Started" step is kicked off).

So no, this doesn't refactor ova-by-tag code in the engine, but rather uses `admiral.DefaultDiscovery.Discover()` to find an ova by tag, and it uses the ova URL to compare the two IP addresses. If that checks out, then the `ManagedBy` info is set on the OVA VM so that the plugin and the OVA can play nice together.

Eventually this will all move into the fileserver code and will run after the "Getting Started" step is complete. The user will not need to re-enter credentials (as they do now by having to run the script), as they will already be known at the time of execution since they are entered at the beginning of "Getting Started". All of this logic will be removed from the vic-ui binary and it will return to a more plugin-agnostic universal installer.